### PR TITLE
Egi filtering- Using addComposing to parse number of passes

### PR DESCRIPTION
**Pull request by kstppd**

### CODE REVIEW

**Comment by kstppd** [comment]
I am thinking that I should use a multimap for the number of passes for each individual ref level. It would make it more foolproof and safer. Opinions? @ykempf  @markusbattarbee @ursg

**Comment by markusbattarbee** [comment]
As per Flowdock discussion, I think a vector would be good. I thought boost program options would have had simple support for a vector of ints as a single value, but not so sure of that...? Found some results on stackoverflow that show convoluted methods to achieve that.

Suggestions:
Do a reverse sort of the blurpass levels so that if the user puts them in L3->L0 order it gets corrected:
    sort(a.begin(), a.end(), greater<int>()); 
This would also fix the potential issues of ordering if using addComposing.

replace P::maxNumPasses with the largest value (i.e. after sorting; P::numPasses.at(0))

Add a verification that P::numPasses.size() is equal to P::amrMaxSpatialRefLevel+1

**Comment by kstppd** [comment]
I think it is good to go now.

File: parameters.cpp [hunk 1]
```diff
@@ -467,7 +471,43 @@ bool Parameters::getParameters(){
...
+   if (!isEmpty){
+
+      for (auto i : blurPassString){
+         P::numPasses.push_back(std::stoi(i));
+      }
+
+      
+      // Reverse Sort and Get the maximum number of filter passes
+      sort(numPasses.begin(), numPasses.end(), greater<int>());
+      P::maxNumPasses = *max_element(P::numPasses.begin(), P::numPasses.end());
```
Comment (kstppd) [inline parameters.cpp:489]: @markusbattarbee It belongs to Parameteres and it is also calculated as the max element from numPasses above.

Comment (markusbattarbee) [reply parameters.cpp:489]: right, well it should be either or. A parameter is an input thing, not something calculated in the code.

Comment (kstppd) [reply parameters.cpp]: Fixed!

**Pull request merged**